sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/ServiceTag.java

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.network.models;
+
+
+import com.azure.core.management.Region;
+import com.azure.core.util.CoreUtils;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+
+/**
+ * A service tag represents a group of IP address prefixes from a given Azure service.
+ * You can use service tags to define network access controls on network security groups, Azure Firewall, and user-defined routes.
+ * Use service tags in place of specific IP addresses when you create security rules and routes.
+ */
+public class ServiceTag {
+    private static final Map<String, ServiceTag> VALUES_BY_NAME = new ConcurrentHashMap<>();
+    /** Action Group. */
+    public static final ServiceTag ACTION_GROUP = new ServiceTag("ActionGroup", true, false, false, false);
+    /** Azure Storage.
+     * <p>Note: This tag represents the service, but not specific instances of the service.
+     * For example, the tag represents the Azure Storage service, but not a specific Azure Storage account.</p>
+     */
+    public static final ServiceTag STORAGE = new ServiceTag("Storage", false, true, true, true);
+    /**
+     * The IP address space that's outside the virtual network and reachable by the public internet.
+     * The address range includes the <a href="https://www.microsoft.com/en-us/download/details.aspx?id=41653">Azure-owned public IP address space</a>.
+     */
+    public static final ServiceTag INTERNET = new ServiceTag("Internet", true, true, false, false);
+    /**
+     * The virtual network address space (all IP address ranges defined for the virtual network),
+     * all connected on-premises address spaces, peered virtual networks, virtual networks connected to a virtual network gateway,
+     * the virtual IP address of the host, and address prefixes used on user-defined routes.
+     * This tag might also contain default routes.
+     */
+    public static final ServiceTag VIRTUAL_NETWORK = new ServiceTag("VirtualNetwork", true, false, false, false);
+
+    /**
+     * Get service tag from name.
+     *
+     * @param serviceTagName the name of the service tag, can be with or without region
+     * @return service tag, null if invalid
+     */
+    public static ServiceTag fromName(String serviceTagName) {
+        if (serviceTagName == null) {
+            return null;
+        }
+        String[] tagParts = serviceTagName.split(Pattern.quote("."));
+        if (tagParts.length == 1) { // root Tag
+            if (serviceTagName.contains(".")) {
+                return null;
+            }
+            return VALUES_BY_NAME.get(tagParts[0]);
```

Comment:
what if `VALUES_BY_NAME` does not have this `serviceTagName`?

Reply:
It'll return `null`. This was on purpose since I enforced the input to be within the constants.

Seems like this enforcement is not necessary, and will add on burden to maintain it to the latest. I'll remove this enforcement.

Reply:
However, if the tag is not within the constant, its property such as `regional`, `inbound` will be undefined. 

I'm thinking removing these properties as they are just indicators.

Reply:
I don't know how often service would add these serviceTag. But prefer it to be extensible, as you have API to return ServiceTag from service response (and user could be on older SDK).

Yes, if you do not sue the `regional` `inbound` (or only use it in runtime check?), we probably can ignore them (we should be able to add later, if required).
Tend to do less client check (especially if service value can change in runtime).

Reply:
There is `ServiceTagsClient.list()`, yet without `regional` or `inbound`. 

I'll remove the properties as they seem useless now.